### PR TITLE
Unsupported country message

### PR DESCRIPTION
**Pull request by tomer-shvadron**
### Description
Added a message on KYB and UBO blocks for unsupported countries

### Screenshots
image

### CODE REVIEW

File: apps/backoffice-v2/src/lib/blocks/hooks/useKybRegistryInfoBlock/useKybRegistryInfoBlock.tsx [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
+import * as React from 'react';
 
-export const useKybRegistryInfoBlock = ({ pluginsOutput, workflow }) => {
-  return useMemo(() => {
-    if (Object.keys(pluginsOutput?.businessInformation?.data?.[0] ?? {}).length === 0) {
-      return [];
-    }
+export const useKybRegistryInfoBlock = ({ pluginsOutput, workflow }) =>
+  useMemo(() => {
+    console.log(pluginsOutput?.businessInformation);
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/hooks/useKybRegistryInfoBlock/useKybRegistryInfoBlock.tsx]: Remove the log.

Comment (tomer-shvadron) [reply apps/backoffice-v2/src/lib/blocks/hooks/useKybRegistryInfoBlock/useKybRegistryInfoBlock.tsx]: Thanks!

File: apps/backoffice-v2/src/lib/blocks/hooks/useUbosBlock/useUbosBlock.tsx [hunk 2]
```diff
@@ -1,30 +1,19 @@
...
-          })
-          .addCell({
-            type: 'subheading',
-            value: 'Registry-provided Data',
-            props: {
-              className: 'mb-4',
-            },
-          })
-          .addCell({
+export const useUbosBlock = (ubos?: Ubo[], message?: string) =>
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/hooks/useUbosBlock/useUbosBlock.tsx]: 100% should do this for ubos, not certain about message. Ubos is not optional it is just something that can be undefined.
```suggestion
export const useUbosBlock = (ubos: Ubo[] | undefined, message: string | undefined) =>
```

File: apps/backoffice-v2/src/lib/blocks/hooks/useKybRegistryInfoBlock/useKybRegistryInfoBlock.tsx [hunk 3]
```diff
@@ -1,11 +1,38 @@
...
+            <span className="mt-[20px] flex text-sm text-black/60">
+              <WarningFilledSvg
+                className={'mr-[8px] mt-px text-black/20'}
+                width={'20'}
+                height={'20'}
+              />
+              <span>{pluginsOutput?.businessInformation?.message}</span>
+            </span>
+          ),
+        };
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/hooks/useKybRegistryInfoBlock/useKybRegistryInfoBlock.tsx]: Feel free to create a type helper or alias. This is only needed until the blocks package is fixed.
```suggestion
    const cell = Object.keys(pluginsOutput?.businessInformation?.data?.[0] ?? {}).length
      ? {
          type: 'details',
          hideSeparator: true,
          value: {
            data: Object.entries(pluginsOutput?.businessInformation?.data?.[0])?.map(
              ([title, value]) => ({
                title,
                value,
              }),
            ),
          },
          workflowId: workflow?.id,
          documents: workflow?.context?.documents,
        } satisfies Extract<Parameters<ReturnType<typeof createBlocksTyped>['addCell']>[0], {
            type: 'details';
          }>
      : {
          type: 'paragraph',
          value: (
            <span className="mt-[20px] flex text-sm text-black/60">
              <WarningFilledSvg
                className={'mr-[8px] mt-px text-black/20'}
                width={'20'}
                height={'20'}
              />
              <span>{pluginsOutput?.businessInformation?.message}</span>
            </span>
          ),
        } satisfies Extract<Parameters<ReturnType<typeof createBlocksTyped>['addCell']>[0], {
            type: 'paragraph';
          }>;
```

File: apps/backoffice-v2/src/lib/blocks/hooks/useUbosBlock/useUbosBlock.tsx [hunk 4]
```diff
@@ -45,17 +34,43 @@ export const useUbosBlock = ubos => {
...
+              <span className="mt-[20px] flex text-sm text-black/60">
+                <WarningFilledSvg
+                  className={'mr-[8px] mt-px text-black/20'}
+                  width={'20'}
+                  height={'20'}
+                />
+                <span>{message}</span>
+              </span>
+            ),
+          };
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/hooks/useUbosBlock/useUbosBlock.tsx]:
```suggestion
          } satisfies Extract<Parameters<ReturnType<typeof createBlocksTyped>['addCell']>[0], {
            type: 'paragraph';
          }>;
```

File: apps/backoffice-v2/src/lib/blocks/hooks/useUbosBlock/useUbosBlock.tsx [hunk 5]
```diff
@@ -45,17 +34,43 @@ export const useUbosBlock = ubos => {
...
               ],
-              data: ubos.map(({ name, percentage, type, level }) => ({
-                name,
-                percentage,
-                type,
-                level,
-              })),
+              data: ubos,
             },
+          }
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/hooks/useUbosBlock/useUbosBlock.tsx]:
```suggestion
          } satisfies Extract<Parameters<ReturnType<typeof createBlocksTyped>['addCell']>[0], {
            type: 'table';
          }>
```

File: apps/backoffice-v2/src/lib/blocks/hooks/useUbosBlock/useUbosBlock.tsx [hunk 6]
```diff
@@ -1,30 +1,19 @@
...
-export const useUbosBlock = ubos => {
-  return useMemo(() => {
-    if (!Array.isArray(ubos) || !ubos?.length) {
-      return [];
-    }
+type Ubo = {
+  name?: string;
+  percentage?: number;
+  type?: string;
+  level: number;
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/hooks/useUbosBlock/useUbosBlock.tsx]: Are all these truly optional are is this supposed to be a union? i.e. `{ level: number; } | { name: string; percentage: number; type: string; level: number; }`

Comment (tomer-shvadron) [reply apps/backoffice-v2/src/lib/blocks/hooks/useUbosBlock/useUbosBlock.tsx]: We don't normalize the data from UBO right now so I want to be on the safe side. I see that I missed one with the `level` property

**Review by Omri-Levy — APPROVED** [review]

**Review by Omri-Levy — APPROVED** [review]

**Pull request merged**